abp168/shadows_of_knight_ep1.py

```diff
@@ -0,0 +1,105 @@
+# *************************************************
+# Dr. Jones Comments - Shadows of the Knight Ep. 1
+# *************************************************
+# This solves the `Shadows of the Knight Ep.1 puzzle <https://www.codingame.com/training/medium/shadows-of-the-knight-episode-1>`_.
+#
+# Problem
+# ===========
+# From the `Shadows of the Knight Ep.1 puzzle <https://www.codingame.com/training/medium/shadows-of-the-knight-episode-1>`_, the problem statement is below:
+#
+# You (Batman) will look for the hostages on a given building by jumping from one window to
+# another using your grapnel gun. Your goal is to jump to the window where the hostages are
+# located in order to disarm the bombs. Unfortunately, you have a limited number of jumps before
+# the bombs go off…
+# Before each jump, the heat-signature device will provide you with the direction of the bombs
+# based on your current position:
+#
+# * U(Up)
+# * UR (Up Right)
+# * R (Right)
+# * DR (Down Right)
+# * DL (Down Left)
+# * L (Left)
+# * UL (Up Left)
+#
+# Your mission is to program the device so that **it indicates the location of the next window you
+# should jump to** in order to reach the bombs' room **as soon as possible**.
+#
+#  
+# Solution Method
+# ===============
+# The following section demonstrate the solution used to solve the Shadows of the Kinght Ep.1 Problem
+#
+# Background
+# ----------
+# The Shadows of the Knight Ep.1 problem can be onsidered as a two dimensional `binary search problem <https://www.geeksforgeeks.org/binary-search/>`_,
+# with the width array being one binary search and the height array being another binary search. I
+# chose to return the next spot for batman as the midpoint of the width and height indices. Below is a diagram of the problem solution
+#
+# .. image:: diagram.png
+#  :width: 100%
+#
+# .. note::
+#
+#   This is very helpful! Explain a bit more with text following the image: why change max_h/w or minh_w by one? How does the formula you came up with implement the binary search?
+#
+# As depctied in the image above, using binary search can help window the target bomb for Batman. If one were to use the min and max boundries in the width and height axis as a guide to either increment or decrement (on each axis respectfully) depending on the provided input direction of the relative bomb location, the bomb could be find in O(logn) time instead of iterating through each position in the respected axis. In order for Batman to find the bomb without going past the target area, it is necessary to have a small step size during the binary search algorithm. As a result, the algorithm should only update ``max_h/w`` and ``min_h/w`` by 1 depending on the relative direction of the bomb to batman.
```

Comment:
Install a spell-checker for VSCode. I use [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker).

Reply:
Installed and fixed spelling errors